pkg/scaffold/constants.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaffold
+
+import (
+	"path/filepath"
+)
+
+const (
+	// file modes
+	defaultDirFileMode  = 0750
+	defaultFileMode     = 0644
+	defaultExecFileMode = 0744
+
+	// Separator to statically create directories.
+	filePathSep = string(filepath.Separator)
+
+	// dirs
+	cmdDir        = "cmd"
+	managerDir    = cmdDir + filePathSep + "manager"
+	pkgDir        = "pkg"
+	apisDir       = pkgDir + filePathSep + "apis"
+	controllerDir = pkgDir + filePathSep + "controller"
+	buildDir      = "build"
+	buildTestDir  = buildDir + filePathSep + "test-framework"
+	deployDir     = "deploy"
+	olmCatalogDir = deployDir + filePathSep + "olm-catalog"
+	crdsDir       = deployDir + filePathSep + "crds"
+	scriptsDir    = "scripts"
+	codegenDir    = scriptsDir + filePathSep + "codegen"
+	versionDir    = "version"
+
+	// files
+	cmdFile                = "main.go"
+	apisFile               = "apis.go"
+	controllerFile         = "controller.go"
+	dockerfileFile         = "Dockerfile"
+	buildFile              = "build.sh"
+	goTestScriptFile       = "go-test.sh"
+	boilerplateFile        = "boilerplate.go.txt"
+	versionFile            = "version.go"
+	docFile                = "doc.go"
+	registerFile           = "register.go"
+	typesFile              = "types.go"
```

Comment:
The reason tests are currently failing is because the current refactor branch has the types as <kind>_types.go instead of just types.go. The scaffolding needs to be updated to properly name the types file.

Reply:
So the correct types file format should be `<kind>_types.go` correct?

Reply:
Yes. So for the `memcached operator`, we have `memcached_types.go`